src/main/java/org/dependencytrack/search/IndexManager.java

```diff
@@ -208,6 +249,19 @@ public void commit() {
         }
     }
 
+    protected void handleCorruptIndexException(CorruptIndexException e) {
+        LOGGER.error("Corrupted Lucene index detected", e);
+        Notification.dispatch(new Notification()
+                .scope(NotificationScope.SYSTEM)
+                .group(NotificationGroup.INDEXING_SERVICE)
+                .title(NotificationConstants.Title.CORE_INDEXING_SERVICES + "(" + indexType.name().toLowerCase() + ")")
+                .content("Corrupted Lucene index detected. Check log for details. " + e.getMessage())
+                .level(NotificationLevel.ERROR)
+        );
+        LOGGER.info("Trying to rebuild the corrupted index " + indexType.name());
+        Event.dispatch(new IndexEvent(IndexEvent.Action.REINDEX, indexType.getClazz()));
```

Comment:
Given rebuilding indexes is an expensive operation, could this cause problems when corrupt indexes are encountered in multiple threads simultaneously (e.g. when multiple internal analysis tasks are running in parallel or close succession)? Reindex events would start piling up and consume significant resources.

Maybe reindexing should be a separate event of type `SingletonCapableEvent`, so that only one instance per `IndexType` can ever be queued? There should be static `chainIdentifier`s per `IndexType` to enable this.

Reply:
Yeah, that's a great idea. Didn't know about `SingletonCapableEvent`

Reply:
Done by leveraging the existing event class and the `IndexEvent.Action` enum to drive the singleton decision.